syntax = "proto3";

import "google/protobuf/timestamp.proto";

option java_multiple_files = true;
option java_package = "com.licscrap.models";
option java_outer_classname = "Models";
option objc_class_prefix = "MDLS";

package models;

service ModelService {
    rpc getPublications(GetRequest) returns (stream Publication){}
    rpc setInterest(MarkRequest) returns (MarkResponse){} // not impl

}

message Publication{
    string name = 1;
    string title = 2;
    string url = 3;
    string description = 4;
    string content = 5;
    string hierarchy = 6;
    // url_source = ForeignKeyField(UrlSource, backref='publications')
    google.protobuf.Timestamp pub_date = 7;
    string art_type = 8;
    float prob = 9;
    bool content_fetched = 10;
    google.protobuf.Timestamp  created_at = 11;
    // google.protobuf.Timestamp s = 4;
    int32 mark = 12;
    int32 status = 13;
    google.protobuf.Timestamp due_date = 14;
    int32 due_reason = 15;
    int32 id = 16;
}


message GetRequest{
    int32 id = 1;
    string type = 2;
    int32 page = 3;
    int32 amount = 4;
}

message MarkRequest{
    int32 id = 1;
    int32 mark = 2;
}
message MarkResponse{
    bool confirmation = 1;
}